network/setup-ip

```diff
@@ -159,6 +159,14 @@ qubes_ip_change_hook() {
 
 have_qubesdb || exit 0
 
+# if called from udev, make sure we don't enable the network before
```

Comment:
This could also be implemented using `SYSTEMD_WANTS` instead.

Reply:
Wouldn't that break the system boot if VM has no network attached?

Reply:
It would not.  `SYSTEMD_WANTS` only queues a start job for a unit.  Instead of having `setup-ip` be called from udev, the udev rule would queue a start job for `qubes-network-uplink.service`.